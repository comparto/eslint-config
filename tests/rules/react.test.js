const react = require('../../src/rules/react')

test('react.js', () => {
  expect(react).toMatchSnapshot()
})
