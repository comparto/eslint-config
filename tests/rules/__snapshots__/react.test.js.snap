// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react.js 1`] = `
Object {
  "extends": Array [
    "plugin:jsx-a11y/recommended",
  ],
  "plugins": Array [
    "react",
    "react-hooks",
    "jsx-a11y",
  ],
  "rules": Object {
    "jsx-a11y/lang": "error",
    "react-hooks/exhaustive-deps": "error",
    "react-hooks/rules-of-hooks": "error",
    "react/boolean-prop-naming": Array [
      "error",
      Object {
        "rule": "^(is|has|should|can|did|will)[A-Z]([A-Za-z0-9]?)+",
      },
    ],
    "react/button-has-type": "error",
    "react/default-props-match-prop-types": "error",
    "react/destructuring-assignment": "off",
    "react/display-name": "warn",
    "react/forbid-component-props": "off",
    "react/forbid-dom-props": "off",
    "react/forbid-elements": "off",
    "react/forbid-foreign-prop-types": "off",
    "react/forbid-prop-types": Array [
      "error",
      Object {
        "checkChildContextTypes": true,
        "checkContextTypes": true,
        "forbid": Array [
          "any",
          "array",
          "object",
        ],
      },
    ],
    "react/function-component-definition": "off",
    "react/jsx-boolean-value": Array [
      "error",
      "never",
    ],
    "react/jsx-child-element-spacing": "off",
    "react/jsx-closing-bracket-location": "off",
    "react/jsx-closing-tag-location": "off",
    "react/jsx-curly-brace-presence": "off",
    "react/jsx-curly-newline": "off",
    "react/jsx-curly-spacing": "off",
    "react/jsx-equals-spacing": "off",
    "react/jsx-filename-extension": "off",
    "react/jsx-first-prop-new-line": "off",
    "react/jsx-fragments": Array [
      "error",
      "syntax",
    ],
    "react/jsx-handler-names": "off",
    "react/jsx-indent": "off",
    "react/jsx-indent-props": "off",
    "react/jsx-key": "warn",
    "react/jsx-max-depth": "off",
    "react/jsx-max-props-per-line": "off",
    "react/jsx-no-bind": "off",
    "react/jsx-no-comment-textnodes": "warn",
    "react/jsx-no-duplicate-props": Array [
      "error",
      Object {
        "ignoreCase": true,
      },
    ],
    "react/jsx-no-literals": "off",
    "react/jsx-no-script-url": "error",
    "react/jsx-no-target-blank": "warn",
    "react/jsx-no-undef": "warn",
    "react/jsx-no-useless-fragment": "error",
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-pascal-case": "error",
    "react/jsx-props-no-multi-spaces": "off",
    "react/jsx-props-no-spreading": "off",
    "react/jsx-sort-default-props": "off",
    "react/jsx-sort-props": "off",
    "react/jsx-tag-spacing": "off",
    "react/jsx-uses-react": "warn",
    "react/jsx-uses-vars": "warn",
    "react/jsx-wrap-multilines": "off",
    "react/no-access-state-in-setstate": "error",
    "react/no-adjacent-inline-elements": "warn",
    "react/no-array-index-key": "off",
    "react/no-children-prop": "warn",
    "react/no-danger": "off",
    "react/no-danger-with-children": "warn",
    "react/no-deprecated": "warn",
    "react/no-did-mount-set-state": "error",
    "react/no-did-update-set-state": "error",
    "react/no-direct-mutation-state": "warn",
    "react/no-find-dom-node": "warn",
    "react/no-is-mounted": "warn",
    "react/no-multi-comp": "off",
    "react/no-redundant-should-component-update": "error",
    "react/no-render-return-value": "warn",
    "react/no-set-state": "off",
    "react/no-string-refs": "warn",
    "react/no-this-in-sfc": "error",
    "react/no-typos": "error",
    "react/no-unescaped-entities": "warn",
    "react/no-unknown-property": "warn",
    "react/no-unsafe": "off",
    "react/no-unused-prop-types": "error",
    "react/no-unused-state": "error",
    "react/no-will-update-set-state": "error",
    "react/prefer-es6-class": "off",
    "react/prefer-read-only-props": "off",
    "react/prefer-stateless-function": "off",
    "react/prop-types": "error",
    "react/react-in-jsx-scope": "warn",
    "react/require-default-props": "off",
    "react/require-optimization": "off",
    "react/require-render-return": "warn",
    "react/self-closing-comp": "error",
    "react/sort-comp": "off",
    "react/sort-prop-types": "off",
    "react/state-in-constructor": "off",
    "react/static-property-placement": "off",
    "react/style-prop-object": "error",
    "react/void-dom-elements-no-children": "error",
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
